core/src/main/java/org/jboss/set/mjolnir/archive/ArchivingBean.java

```diff
@@ -15,26 +15,41 @@
     private Logger logger = Logger.getLogger(getClass());
 
     private UsernamePasswordCredentialsProvider credentialsProvider;
+    private Configuration configuration;
 
     @Inject
-    public ArchivingBean(UsernamePasswordCredentialsProvider credentialsProvider) {
-        this.credentialsProvider = credentialsProvider;
+    public ArchivingBean(Configuration configuration) {
+        this.configuration = configuration;
+        this.credentialsProvider = new UsernamePasswordCredentialsProvider("token", configuration.getGitHubToken());
     }
 
     /**
      * Create remote repository for given repository object
      *
      * @param repository object representing repository for archiving
-     * @return void
      */
-    public void createRepositoryMirror(String organization, Repository repository) throws GitAPIException, URISyntaxException {
-        logger.infof("Archiving %s", repository.getCloneUrl());
-        /** TODO
-         * get parent directly from object
-         */
-        String parentUrl = repository.getCloneUrl().replace(repository.getOwner().getLogin(), organization);
-
-        File organizationDirectory = new File(System.getProperty("user.home"), organization);
+    public void createRepositoryMirror(Repository repository) throws GitAPIException, URISyntaxException, IOException {
+        logger.infof("Archiving repository %s", repository.getCloneUrl());
+
+        if (repository.getSource() == null || repository.getSource().getOwner() == null) {
+            throw new IllegalArgumentException("Source repository and it's owner has to be set.");
+        }
+
+        String parentUrl = repository.getSource().getCloneUrl();
+
+        File archiveRoot;
+        if (configuration.getRepositoryArchiveRoot() != null) {
+            archiveRoot = new File(configuration.getRepositoryArchiveRoot());
+        } else {
+            archiveRoot = new File(System.getProperty("user.home"));
+        }
+
+        String sourceOrganizationName = repository.getSource().getOwner().getLogin();
+        if (sourceOrganizationName == null) {
```

Comment:
The condition should test `!StringUtils.isBlank(sourceOrganizationName)` instead of just null check. Will fix this. Need to import apache commons-utils.